pkg/cmd/roachtest/cluster.go

```diff
@@ -1585,6 +1585,59 @@ func (c *clusterImpl) FetchCores(ctx context.Context, l *logger.Logger) error {
 	})
 }
 
+// FetchPebbleCheckpoints fetches any Pebble checkpoints on the cluster. These
+// are typically generated by failed consistency checks, i.e. replica
+// divergence, and only contain data for the inconsistent ranges.
+func (c *clusterImpl) FetchPebbleCheckpoints(ctx context.Context, l *logger.Logger) error {
+	// Don't grab checkpoints on empty clusters or local runs.
+	if c.spec.NodeCount == 0 || c.IsLocal() {
+		return nil
+	}
+
+	// Checkpoints can be large. Bail out after 3 minutes.
+	return timeutil.RunWithTimeout(ctx, "checkpoints", 3*time.Minute, func(ctx context.Context) error {
+		numStores := 1
+		if ssds := c.spec.SSDs; ssds > 1 {
+			numStores = ssds
+		}
+		for storeIdx := 1; storeIdx <= numStores; storeIdx++ {
+			// Find any checkpoints.
+			//
+			// TODO(erikgrinaker): expose and use SyncedCluster.NodeDir(). We can't
+			// use the {store-dir} expansion which only refers to the first store.
```

Comment:
Another option would be for `{store-dir}` to take a store index (more consistent with how similar functionality is exposed).

Reply:
Yeah, may as well -- I'll have a look.

Reply:
Done.